src/DotNetEnv/Parsers.cs

```diff
@@ -176,14 +177,16 @@ private static string ToUtf32Char (string hex)
         // unquoted values can have interpolated variables,
         // but only inline whitespace -- until a comment,
         // and no escaped chars, nor byte code chars
-        // FIXME: would be nice to solve multi word with comment case with parser directly (no split + trim)
         internal static readonly Parser<ValueCalculator> UnquotedValue =
-            InterpolatedValue.Or(
-                NotControlNorWhitespace("'\"$")
-                .Or(InlineWhitespaceChars.AtLeastOnce().Text())
-                .AtLeastOnce()
-                .Select(strs => new ValueActual(strs))
-            ).Many().Select(vs => new ValueCalculator(vs).Split("[ \t]#").Trim());
+            Parse.Regex("[ \t\"']").Not()
+                .Then(_ =>
+                    InterpolatedValue
+                        .Or(InlineWhitespace.Then(inlineWhitespaces
+                            => Parse.Char('#').Not()
+                                .Then(_2 => NotControlNorWhitespace("$\"'"))
+                                .Select(valueString => (IValue)new ValueActual(string.Concat(inlineWhitespaces, valueString)))))
```

Comment:
The cast should be unnecessary, and sprache has concat we should use at this point.

Reply:
Yes, cast was remaining from before, I will remove it.

But I'm not sure what you mean regarding concat.
I had a look into your branch, and there you use concat from Linq, but that's because you work on char-Arrays and convert to a string by using `Text` afterwards.
Here we have strings, not char-Arrays, and we expect a string as response.
I can use concat on the strings of course, but that will return a char-Array, which I need to convert to a string afterwards.

If there is something different I just don't see, please give me some hint, or a code-suggestion.

Reply:
Updated, leaving it open for your response about concat

Reply:
Ah, indeed, I was uncertain what types you had at this point. I'm fine resolving this thread, as at this point I'm more interested in your answer to: https://github.com/tonerdo/dotnet-env/pull/82#discussion_r1432311810